mesonbuild/interpreter/interpreter.py

```diff
@@ -878,6 +877,10 @@ def do_subproject(self, subp_name: str, kwargs: kwtypes.DoSubproject, force_meth
             return self.disabled_subproject(subp_name, disabled_feature=feature)
 
         default_options = kwargs['default_options']
+        if isinstance(default_options, str):
+            default_options = [default_options]
+        if isinstance(default_options, list):
+            default_options = dict((x.split('=', 1) for x in default_options))
```

Comment:
Instead of doing this, could we just revert the change that took the validation and conversion out of the `typed_kwargs` call? I really never understood why we wanted to leak DSL details into the IR anyway.

I can live with this if it makes the backport easier, but in the long run it's the `typed_kwargs` job to handle this.

Reply:
I agree that we want to get there but the way is messy. With #14559 I'm trying to at least make OptionStore accept dictionaries, I'll try to put on top the change to handle validation and conversion in the interpreter.